src/ops/phoenix_ops.rs

```diff
@@ -1,110 +1,107 @@
 use super::AbiCall;
-use crate::call_context::{CallContext, Resolver};
+use crate::call_context::{ArgsExt, CallContext, Resolver};
 use crate::VMError;
-// use std::convert::{TryFrom, TryInto};
-// use std::env;
-// use std::path::Path;
-
+use dusk_abi::PodExt;
 use kelvin::ByteHash;
-// use phoenix::{
-//     db, utils, zk, BlsScalar, NoteGenerator, NoteVariant, PublicKey,
-//     Transaction, TransactionInput, TransactionOutput, TransparentNote,
-// };
-// use phoenix_abi::{Input, Note, Proof};
+use phoenix::{
+    db, utils, zk, BlsScalar, NoteGenerator, NoteVariant, PublicKey,
+    Transaction, TransactionInput, TransactionOutput, TransparentNote,
+};
+use phoenix_abi::{Input, Note, Proof};
+use std::convert::{TryFrom, TryInto};
+use std::env;
+use std::path::Path;
 use wasmi::{RuntimeArgs, RuntimeValue, ValueType};
 
-// const SUCCESS: Result<Option<RuntimeValue>, VMError> =
-//     Ok(Some(RuntimeValue::I32(1)));
+const SUCCESS: Result<Option<RuntimeValue>, VMError> =
+    Ok(Some(RuntimeValue::I32(1)));
 
-// const FAIL: Result<Option<RuntimeValue>, VMError> =
-//     Ok(Some(RuntimeValue::I32(0)));
+const FAIL: Result<Option<RuntimeValue>, VMError> =
+    Ok(Some(RuntimeValue::I32(0)));
 
 pub struct PhoenixStore;
 
-// #[inline]
-// fn has_value(bytes: &[u8]) -> bool {
-//     !bytes.iter().all(|b| *b == 0)
-// }
+#[inline]
+fn has_value(bytes: &[u8]) -> bool {
+    !bytes.iter().all(|b| *b == 0)
+}
+
+fn get_tx<S, H>(
+    context: &mut CallContext<S, H>,
+    args: RuntimeArgs,
+) -> Result<Transaction, VMError>
+where
+    S: Resolver<H>,
+    H: ByteHash,
+{
+    let inputs_ptr = args.get(0)? as usize;
+    let notes_ptr = args.get(1)? as usize;
+    let proof_ptr = args.get(2)? as usize;
+
+    let (inputs_buf, notes_buf, proof_buf): (Vec<_>, Vec<_>, Vec<_>) = context
```

Comment:
You probably could use (&[u8], &[u8], &[u8]) here for efficiency, but it's something we can optimize later.